### PR TITLE
Restaurant registration

### PR DESCRIPTION
**Pull request by casperjureskog**
**PT Story:** https://www.pivotaltracker.com/story/show/142510605

**Changes proposed in this PR:**

* Adds restaurant view
* Adds restaurant controller
* Adds restaurant DataBase

**What I have learned working on this feature:**
Its not easy to constrict the routes
and check spelling an extra time.

**Screenshots:**

screen shot 2017-03-28 at 21 07 44

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resource :restaurant
```
Comment (tochman) [inline config/routes.rb]: it's `resources` and it's also `:restaurants` 😕

File: app/controllers/restaurants_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  def new
+    @restaurants = Restaurant.new
+  end
+
+  def create
+    @restaurants = Restaurant.new restaurant_params
+
+    if @restaurants.save
+      flash[:notice] = "Restaurant successfully created"
```
Comment (tochman) [inline app/controllers/restaurants_controller.rb]: Why redirect and not render?

File: spec/models/restaurant_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Restaurant, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (tochman) [inline spec/models/restaurant_spec.rb]: There should be specs here. Remember to stick to the acceptance unit test cycle. 🤔

File: config/routes.rb [hunk 4]
```diff
@@ -1,3 +1,3 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :restaurants
```
Comment (tochman) [inline config/routes.rb]: Remember that you can (and should) scope the routes you create using the 'only:' keyword

File: spec/models/restaurant_spec.rb (hunk 3 above)
Comment (casperjureskog) [reply spec/models/restaurant_spec.rb]: yes we are working on it :-) we did remember after our cucumber went thru

File: app/controllers/restaurants_controller.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+class RestaurantsController < ApplicationController
+
+  def new
+    @restaurants = Restaurant.new
```
Comment (tochman) [inline app/controllers/restaurants_controller.rb]: Should be singular

File: app/controllers/restaurants_controller.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def create
+    @restaurants = Restaurant.new restaurant_params
```
Comment (tochman) [inline app/controllers/restaurants_controller.rb]: Same here. The plural form should be used for a collection. I.e. In the index method (if you create one later)

File: app/views/restaurants/new.html.haml [hunk 7]
```diff
@@ -0,0 +1,28 @@
+%h2 Restaurant registration
+
+= form_for (@restaurants) do |f|
```
Comment (tochman) [inline app/views/restaurants/new.html.haml]: Same here @restaurant.

**Comment by fottander** [comment]
@tochman Can´t get the rspec tests to work, what are we doing wrong? 
Solved it!

File: app/models/restaurant.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+class Restaurant < ApplicationRecord
+  validates :name, presence: true
+  validates :street, presence: true
+  validates :zip_code, presence: true
+  validates :city, presence: true
+  validates :food_style, presence: true
+end
```
Comment (AmberWilkie) [inline app/models/restaurant.rb:7]: Not a blocker, I just want you guys to know there's another way to write this:
`validates_presence_of :name`, etc.

**Review by AmberWilkie — APPROVED** [review]

File: app/models/restaurant.rb (hunk 8 above)
Comment (casperjureskog) [reply app/models/restaurant.rb:7]: yes we talk about that alternative but we like't this better because it look't more readable to us

**Pull request merged**